lib/index.js

```diff
@@ -745,3 +731,22 @@ function _dropCollections(callback) {
     }, callback);
   });
 }
+
+function _createUrl(config) {
+  let url = 'mongodb://';
+  if(config.username) {
+    url += `${config.username}:${config.password}@`;
+  }
+  url += `${config.host}:${config.port}/${config.name}`;
+  // this needs to come last
+  if(config.username) {
+    url +=
+      `?authSource=${config.connectOptions.authSource || 'admin'}`;
+  }
+  return url;
+}
+
+function _sanitizeUrl(url) {
+  const urlParts = require('url').parse(url);
```

Comment:
Looking forward to when this and every other lib is converted to ESM syntax.  Please do not add CommonJS require into functions like this.  Pull in `url` at the top of the file along with all other deps which will make it easier to convert later.

Reply:
Updated - 08b9847